src/code/middleware/firebase-decorator.ts

```diff
@@ -23,11 +31,13 @@ export const Firebasify = (model:any, relativeDataPath:string, callBack?:() => v
         presenceRef.update(presence);
       }
     }
-    ref.once('value').then( (newV:firebase.database.DataSnapshot) => {
+    let onValueCount = 0;
+    ref.on('value', (newV:firebase.database.DataSnapshot) => {
```

Comment:
In the code samples I have seen, its common to first request the value, and when the response comes back, register the value listener.  That pattern made sense to me.

This seems functionally equivalent (assuming that the first 'value' event will be triggered automatically with the current value when we register our dataRef).  

I would be curious why you chose this change.

Reply:
Since the 'on' handler always responds immediately with an update, the previous pattern results in a double-initial update. Since I was putting in place an update counter to fix one of the bugs, the double-update stood out to me.

Reply:
Right -- I remember that now.  Thanks